File: kotlin/kindergarten-garden/src/main/kotlin/KindergartenGarden.kt
class KindergartenGarden(private val diagram: String) {

    private val rows = diagram.split("\n")

    private val students = listOf(
        "Alice", "Bob", "Charlie", "David",
        "Eve", "Fred", "Ginny", "Harriet",
        "Ileana", "Joseph", "Kincaid", "Larry"
    )

    fun getPlantsOfStudent(student: String): List<String> {
        val i = students.indexOf(student)
        require(i != -1) { "Unknown student" }

        val start = i * 2

        return listOf(
            plant(rows[0][start]),
            plant(rows[0][start + 1]),
            plant(rows[1][start]),
            plant(rows[1][start + 1]),
        )
    }

    private fun plant(ch: Char): String = when (ch) {
        'G' -> "grass"
        'C' -> "clover"
        'R' -> "radishes"
        'V' -> "violets"
        else -> error("Unexpected character: $ch")
    }
}

File: csharp/kindergarten-garden/KindergartenGarden.cs
using System.Data;

public enum Plant
{
    Violets,
    Radishes,
    Clover,
    Grass
}

public class KindergartenGarden(string diagram)
{
    private readonly string[] rows = diagram.Split('\n');
    private readonly string[] students = [
        "Alice", "Bob", "Charlie", "David",
        "Eve", "Fred", "Ginny", "Harriet",
        "Ileana", "Joseph", "Kincaid", "Larry"
    ];

    public IEnumerable<Plant> Plants(string student)
    {
        var i = Array.IndexOf(students, student);

        if (i == -1)
        {
            throw new ArgumentException("Unknown student");
        }

        var start = i * 2;


        yield return GetPlant(rows[0][start]);
        yield return GetPlant(rows[0][start + 1]);
        yield return GetPlant(rows[1][start]);
        yield return GetPlant(rows[1][start + 1]);
    }

    private Plant GetPlant(char ch) => ch switch
    {
        'G' => Plant.Grass,
        'C' => Plant.Clover,
        'R' => Plant.Radishes,
        'V' => Plant.Violets,
        _ => throw new ArgumentException("Unexpected character: " + ch)
    };
}